src/config.ts

```diff
@@ -24,16 +27,70 @@ function configOrder(obj: configObjects): configObjects {
   return obj
 }
 
-export function getDefaultShell(): string {
+// finds the default shell start commmand
+export async function getDefaultShell(): Promise<string> {
+  let shellStartCommand
+  if (process.platform === "win32") {
+    // Windows
+    try {
+      shellStartCommand = await which("pwsh.exe")
+      return shellStartCommand
+    } catch (e1) {
+      try {
+        shellStartCommand = await which("powershell.exe")
+        return shellStartCommand
+      } catch (e2) {
+        shellStartCommand = process.env.COMSPEC || "cmd.exe"
+        return shellStartCommand
+      }
+    }
+  } else {
+    // Unix
+    shellStartCommand = process.env.SHELL || "/bin/sh"
+    return shellStartCommand
+  }
+}
+
+// The shell command used in case getDefaultShell does not find a prefered shell
+export function getFallbackShell() {
   return process.platform === "win32" ? process.env.COMSPEC || "cmd.exe" : process.env.SHELL || "/bin/sh"
 }
 
+// set start command asyncronously
+export async function setShellStartCommand() {
+  // If set, automatically detect the prefered shell start command in the next Atom restart.
+  // (it will switched off if you edit `shell` option manually.
+  let shouldAutoShell = localStorage.getItem("terminal.autoShell")
+
+  // set for the first time
+  if (!shouldAutoShell) {
+    localStorage.setItem("terminal.autoShell", "true")
+    shouldAutoShell = "true"
+  }
+
+  if (shouldAutoShell === "true") {
+    // first check the cache
+    const defaultSellCache = localStorage.getItem("terminal.defaultSellCache")
```

Comment:
```diff
-defaultSellCache
+defaultShellCache
```

Reply:
Fixed in https://github.com/atom-ide-community/terminal/pull/27/commits/151d6b2640a291dff0f498c2ef65f3976b31dda8